### PR TITLE
test(slide-switch): add comprehensive unit tests for SlideSwitchHelper

### LINKED ISSUE #300: Write Unit Tests for SlideSwitchHelper Class
**Description:**
  The SlideSwitchHelper class is responsible for managing a slide switch's state and its event listeners. To ensure its reliability and proper functionality, we need to create a comprehensive suite of unit tests.

**Tasks:**

1. Test Initialization Logic:
 Verify that the isOn field is correctly set based on the initial state of the DigitalInput object passed to the constructor.
2. Test Event Listener Management:
  Confirm that listeners are properly added using the addEventListener method.
  Validate that listeners are removed using the removeEventListener method.
  Ensure that the initialize method correctly updates the isOn field when the state of the slide switch changes.
3. Test Logging:
  Verify that the appropriate log messages are generated during the initialization process.
4. Mocking Dependencies:
  Use a mocking framework (e.g., Mockito) to mock the DigitalInput and DigitalStateChangeListener dependencies.
5. Edge Cases:
 Test how the class handles scenarios where no listeners are attached but the switch state changes.

**Acceptance Criteria**
- [ ] All test cases outlined in the tasks above are implemented.
- [ ] Code coverage for the SlideSwitchHelper class is at least 70%.
- [ ] Tests are written using a Java JUnit testing framework
- [ ] Mocking is implemented using a framework such as Mockito or a similar tool.
- [ ] All tests pass successfully, and edge cases are handled gracefully.

### PR DESCRIPTION
**Pull request by Ed0827**
Fixes #300 

Description:
This pull request adds comprehensive unit tests for the SlideSwitchHelper class to verify its functionality and robustness. The tests ensure that the helper correctly initializes its state, manages event listeners, and outputs the expected log messages.

What was changed:

Initialization Tests:

Added tests to verify that the isOn field is correctly set when the DigitalInput is high or low (testIsOnFieldSetCorrectlyWhenInputIsHigh and testIsOnFieldSetCorrectlyWhenInputIsLow).
Event Listener Management:

Implemented tests to confirm that the addEventListener and removeEventListener methods delegate correctly to the underlying DigitalInput (testAddEventListener and testRemoveEventListener).
Created a test (testCustomListenerInvocation) to ensure that custom event listeners are properly invoked.
Added a test (testInternalListenerUpdatesIsOn) that uses an ArgumentCaptor to capture the internal listener registered during initialization and simulates state changes to verify that the isOn field is updated accordingly.
Logging Verification:

Developed a logging test (testInitializationLogsCaptureSystemOut) that captures the output from System.out to assert that the log message "Initializing Slide Switch" is generated during initialization.
Mocking Dependencies:

Used Mockito to mock DigitalInput and DigitalStateChangeListener dependencies, allowing simulation of different input states and verifying listener behavior without actual hardware.
Why was it changed:

These changes are necessary to ensure that the slide switch functionality is robust and reliable, particularly because it directly interacts with hardware via Pi4J.
Verifying the proper initialization, listener management, and logging behavior increases confidence in the codebase before integrating with Raspberry Pi hardware.
The improvements in test coverage (over 70%) help safeguard against regressions and provide a solid foundation for further development.
How was it changed:

JUnit 5 & Mockito: The tests were implemented using JUnit 5 for structure and assertions, and Mockito for mocking dependencies.
State Verification: The tests simulate different DigitalInput states to verify that isOn is set correctly at construction and when events occur.
Listener Testing: By capturing and invoking the internal event listener, the tests ensure that the helper correctly updates its state in response to input changes.
Logging Capture: Instead of modifying dependencies or production code, the test redirects System.out to a ByteArrayOutputStream to capture and verify that the expected log message is output during initialization.

### CODE REVIEW

File: pi4micronaut-utils/src/test/java/com/opensourcewithslu/inputdevices/SlideSwitchHelperTest.java [hunk 1]
```diff
@@ -0,0 +1,94 @@
...
+import com.pi4j.io.gpio.digital.DigitalStateChangeListener;
+import java.util.concurrent.atomic.AtomicBoolean;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.slf4j.Logger;
+
+import static org.junit.jupiter.api.Assertions.*;
+import static org.mockito.Mockito.*;
+
+class SlideSwitchHelperTest {
```
Comment (RalphTan37) [inline pi4micronaut-utils/src/test/java/com/opensourcewithslu/inputdevices/SlideSwitchHelperTest.java:13]: Your tests look very well-structured with great use of instantiations and comments.

**Review by RalphTan37 — COMMENTED** [review]
Great work Seyun!

**Review by yrlmanoharreddy — CHANGES_REQUESTED** [review]
I noticed that changes were made to the actual SlideSwitchHelper class. Please avoid modifying the core helper classes unless absolutely necessary. Kindly revert the changes made to the SlideSwitchHelper class and adjust the test class accordingly. This will help maintain the integrity of the original functionality and ensure that tests are focused on verifying behavior rather than altering the helper class itself. Thanks for your attention to this!

**Review by yrlmanoharreddy — APPROVED** [review]
The SlideSwitchHelper implementation and tests are well-executed. The test cases cover the core functionality effectively, and the use of mocks is appropriate, so this PR is approved. Great work!
